libtock/services/alarm.c

```diff
@@ -7,6 +7,95 @@ static int within_range(uint32_t a, uint32_t b, uint32_t c) {
   return (b - a) < (b - c);
 }
 
+/** \brief Convert milliseconds to clock ticks
+ *
+ * WARNING: This function will silently overflow if the output
+ * number of ticks is larger than `UINT32_MAX`.
```

Comment:
Perhaps it would be better to pass a `uint32_t` reference that this function updates and instead returns a returncode (returning an error on an overflow)? This is not an optimal fix, but is a step in the right direction. Silent overflows have made debugging other parts of the libtock alarm infrastructure challenging and something I think we should avoid when possible.

Reply:
One solution is to check if the calculated ticks value is greater than `UINT32_MAX` prior to casting from `uint64_t` to `uint32_t` and return the relevant `returncode_t` accordingly.

Reply:
I'm not fully sold on the need for returncodes for internal helper functions. What would the code do if it does detect an overflow? This is certainly a step in the right direction, so I have added an assert.

Reply:
I think having the assertion here is enough. I think we can now remove the comment warning of silently overflowing if ticks is greater than 2**32.

Reply:
Who is supposed to do the check to avoid the assert? The application?

Reply:
If I understand correctly, this is only a helper method and is not exposed to applications. The user of this method will be `timer_in` which with #395 will add the logic needed to avoid the assert.

For this internal method, it seems reasonable to leave this assert since we should be avoiding this case.

Reply:
Got it